### PR TITLE
[CARBONDATA-2274] fix for Partition table having more than 4 column giving zero record

### PR DESCRIPTION
**Pull request by rahulforallp**
**Converting of Array[String,String] to Map[String, String] was giving wrong order of partition column. And we were using that sequence to create path.**

**So used LinkedHashMap to avoid reordering.**

 - [X] Any interfaces changed? NO
 
 - [x] Any backward compatibility impacted? No
 
 - [x] Document update required? No

 - [x] Testing done **UT added**
       
 - [x] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA. NA

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3316/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4544/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4041/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4042/

**Comment by CarbonDataQA** [comment]
Build Failed with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3371/

**Comment by CarbonDataQA** [comment]
Build Failed  with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4597/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4094/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4601/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3377/

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4101/

**Comment by rahulforallp** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Fail , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4105/

**Comment by rahulforallp** [comment]
retest sdv please

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4106/

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CarbonScalaUtil.scala [hunk 1]
```diff
@@ -340,12 +342,12 @@ object CarbonScalaUtil {
    * @return updated partition spec
    */
   def updatePartitions(
-      partitionSpec: Map[String, String],
-  table: CarbonTable): Map[String, String] = {
+      partitionSpec: mutable.LinkedHashMap[String, String],
+  table: CarbonTable): mutable.LinkedHashMap[String, String] = {
```
Comment (ravipesala) [inline integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CarbonScalaUtil.scala]: Formatting is not proper

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/standardpartition/StandardPartitionTableLoadingTestCase.scala [hunk 2]
```diff
@@ -135,6 +135,23 @@ class StandardPartitionTableLoadingTestCase extends QueryTest with BeforeAndAfte
...
+        |  projectcode int, projectjoindate Timestamp, projectenddate Timestamp,attendance int)
+        | PARTITIONED BY (utilization int,salary int,workgroupcategory int, empname String,
+        | designation String)
+        | STORED BY 'org.apache.carbondata.format'
+      """.stripMargin)
+    sql(s"""LOAD DATA local inpath '$resourcesPath/data.csv' INTO TABLE partitionfive OPTIONS('DELIMITER'= ',', 'QUOTECHAR'= '"')""")
+
+    validateDataFiles("default_partitionfive", "0", 10)
+
+    checkAnswer(sql("select empno, empname, designation, doj, workgroupcategory, workgroupcategoryname, deptno, deptname, projectcode, projectjoindate, projectenddate, attendance, utilization, salary from partitionfive order by empno"),
```
Comment (ravipesala) [inline integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/standardpartition/StandardPartitionTableLoadingTestCase.scala]: @rahulforallp  If this is about partitions are going to wrong folder then better add some filter test case to reproduce and fix it.

Comment (rahulforallp) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/standardpartition/StandardPartitionTableLoadingTestCase.scala]: @ravipesala you mean i should add filter query also in test case? 
without filter also i was able to reproduce it.

Comment (ravipesala) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/standardpartition/StandardPartitionTableLoadingTestCase.scala]: ok

File: integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CarbonScalaUtil.scala (hunk 1 above)
Comment (rahulforallp) [reply integration/spark-common/src/main/scala/org/apache/carbondata/spark/util/CarbonScalaUtil.scala]: formatting done

File: integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/standardpartition/StandardPartitionTableLoadingTestCase.scala (hunk 2 above)
Comment (rahulforallp) [reply integration/spark-common-test/src/test/scala/org/apache/carbondata/spark/testsuite/standardpartition/StandardPartitionTableLoadingTestCase.scala]: filter test case also added

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder1/4634/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://88.99.58.216:8080/job/ApacheCarbonPRBuilder/3409/

**Comment by ravipesala** [comment]
SDV Build Success , Please check CI http://144.76.159.231:8080/job/ApacheSDVTests/4142/

**Comment by ravipesala** [comment]
LGTM

**Pull request closed without merging**